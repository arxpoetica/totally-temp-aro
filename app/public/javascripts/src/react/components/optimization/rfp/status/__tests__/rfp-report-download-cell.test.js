/* global jest test expect */
import React from 'react'
import { shallow } from 'enzyme'
import RfpReportDownloadCell from '../rfp-report-download-cell.jsx'

const PLAN_ID = 42
const reportDefinitions = [
  {
    reportData: {
      id: 13,
      reportType: 'COVERAGE',
      name: 'site_area_coverage',
      displayName: 'Site Coverage',
      media_types: [
        'csv',
        'json',
        'xls'
      ]
    },
    href: '/rfp/{planId}/report/coverage/13.{mediaType}'
  },
  {
    reportData: {
      id: 14,
      reportType: 'RFP',
      name: 'rfp_report',
      displayName: 'RFP Report',
      media_types: [
        'csv',
        'json',
        'xls'
      ]
    },
    href: '/rfp/{planId}/report/general/14.{mediaType}'
  }
]

// -----------------------------------------------------------------------------
test('Default component render', () => {
  // Mock the Date.now() function so that the test is determinstic
  Date.now = jest.fn(() => 1561358770689)
  const component = shallow(
    <RfpReportDownloadCell
      planId={PLAN_ID}
      reportDefinitions={reportDefinitions}
      userId={1}
    />
  )
  expect(component).toMatchSnapshot()
})

// -----------------------------------------------------------------------------
test('When selected report changes', () => {
  // Mock the Date.now() function so that the test is determinstic
  Date.now = jest.fn(() => 1561358770689)
  const component = shallow(
    <RfpReportDownloadCell
      planId={PLAN_ID}
      reportDefinitions={reportDefinitions}
      userId={1}
    />
  )
  expect(component).toMatchSnapshot()
  component.find('#selectRfpReportDefinition').simulate('change', { target: { value: reportDefinitions[1].reportData.id } })
  expect(component).toMatchSnapshot()
})
