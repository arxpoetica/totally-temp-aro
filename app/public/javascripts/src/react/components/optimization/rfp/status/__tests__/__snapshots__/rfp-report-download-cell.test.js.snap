// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default component render 1`] = `
<div
  className="d-flex"
>
  <select
    className="form-control"
    id="selectRfpReportDefinition"
    onChange={[Function]}
    style={
      Object {
        "marginTop": "1px",
      }
    }
    value={13}
  >
    <option
      key="13"
      value={13}
    >
      Site Coverage
    </option>
    <option
      key="14"
      value={14}
    >
      RFP Report
    </option>
  </select>
  <div
    className="btn btn-group p-0"
  >
    <a
      className="btn btn-light"
      download={true}
      href="/service-download-file/2019-06-24_site_area_coverage.csv/rfp/42/report/coverage/13.csv"
      key="csv"
      style={
        Object {
          "whiteSpace": "nowrap",
        }
      }
    >
      <i
        className="fa fa-download"
      />
       
      csv
    </a>
    <a
      className="btn btn-light"
      download={true}
      href="/service-download-file/2019-06-24_site_area_coverage.json/rfp/42/report/coverage/13.json"
      key="json"
      style={
        Object {
          "whiteSpace": "nowrap",
        }
      }
    >
      <i
        className="fa fa-download"
      />
       
      json
    </a>
    <a
      className="btn btn-light"
      download={true}
      href="/service-download-file/2019-06-24_site_area_coverage.xls/rfp/42/report/coverage/13.xls"
      key="xls"
      style={
        Object {
          "whiteSpace": "nowrap",
        }
      }
    >
      <i
        className="fa fa-download"
      />
       
      xls
    </a>
  </div>
</div>
`;

exports[`When selected report changes 1`] = `
<div
  className="d-flex"
>
  <select
    className="form-control"
    id="selectRfpReportDefinition"
    onChange={[Function]}
    style={
      Object {
        "marginTop": "1px",
      }
    }
    value={13}
  >
    <option
      key="13"
      value={13}
    >
      Site Coverage
    </option>
    <option
      key="14"
      value={14}
    >
      RFP Report
    </option>
  </select>
  <div
    className="btn btn-group p-0"
  >
    <a
      className="btn btn-light"
      download={true}
      href="/service-download-file/2019-06-24_site_area_coverage.csv/rfp/42/report/coverage/13.csv"
      key="csv"
      style={
        Object {
          "whiteSpace": "nowrap",
        }
      }
    >
      <i
        className="fa fa-download"
      />
       
      csv
    </a>
    <a
      className="btn btn-light"
      download={true}
      href="/service-download-file/2019-06-24_site_area_coverage.json/rfp/42/report/coverage/13.json"
      key="json"
      style={
        Object {
          "whiteSpace": "nowrap",
        }
      }
    >
      <i
        className="fa fa-download"
      />
       
      json
    </a>
    <a
      className="btn btn-light"
      download={true}
      href="/service-download-file/2019-06-24_site_area_coverage.xls/rfp/42/report/coverage/13.xls"
      key="xls"
      style={
        Object {
          "whiteSpace": "nowrap",
        }
      }
    >
      <i
        className="fa fa-download"
      />
       
      xls
    </a>
  </div>
</div>
`;

exports[`When selected report changes 2`] = `
<div
  className="d-flex"
>
  <select
    className="form-control"
    id="selectRfpReportDefinition"
    onChange={[Function]}
    style={
      Object {
        "marginTop": "1px",
      }
    }
    value={14}
  >
    <option
      key="13"
      value={13}
    >
      Site Coverage
    </option>
    <option
      key="14"
      value={14}
    >
      RFP Report
    </option>
  </select>
  <div
    className="btn btn-group p-0"
  >
    <a
      className="btn btn-light"
      download={true}
      href="/service-download-file/2019-06-24_rfp_report.csv/rfp/42/report/general/14.csv"
      key="csv"
      style={
        Object {
          "whiteSpace": "nowrap",
        }
      }
    >
      <i
        className="fa fa-download"
      />
       
      csv
    </a>
    <a
      className="btn btn-light"
      download={true}
      href="/service-download-file/2019-06-24_rfp_report.json/rfp/42/report/general/14.json"
      key="json"
      style={
        Object {
          "whiteSpace": "nowrap",
        }
      }
    >
      <i
        className="fa fa-download"
      />
       
      json
    </a>
    <a
      className="btn btn-light"
      download={true}
      href="/service-download-file/2019-06-24_rfp_report.xls/rfp/42/report/general/14.xls"
      key="xls"
      style={
        Object {
          "whiteSpace": "nowrap",
        }
      }
    >
      <i
        className="fa fa-download"
      />
       
      xls
    </a>
  </div>
</div>
`;
